README.md

```diff
@@ -1,32 +1,506 @@
+        <h2>{name}</h2>
+        <p>
+          <Link to={`/profile/${id}`}>View profile</Link>
+        </p>
+    </article>
+  );
+};
+
+export default UserCard;
+```
+
+We should be seeing our array of four users being logged to our browser console.
+
+Instead of passing props from `Home` to `UserCard`, we can just use the
+`useOutletContext` hook to directly access the data that was originally passed
+to our `Outlet` component in `App`. This is a very helpful feature if you ever
+need to pass data to a deeply nested component, and is a great reason to use
+Context Providers in general with React's `useContext` hook.
+
+However, there is a small hitch that we run into if we have deeply nested
+routes.
+
+### useOutletContext and Deeply Nested Routes
+
+If we look at our `routes` in our `routes.js` file, we'll see that we have a
+deeply nested route:
+
+```jsx
+// routes.js
+// ...import statements
+
+const routes = [
+    {
+        path: "/",
+        element: <App />,
+        errorElement: <ErrorPage />,
+        children: [
+             {
+                path: "/",
+                element: <Home />,
+                children: [
+                    {
+                        path: "/profile/:id",
+                        element: <UserProfile />
+                    }
+                ]
+            }, 
+            {
+                path: "/about",
+                element: <About />
+            },
+            {
+                path: "/login",
+                element: <Login />
+            }
+        ]
+    }
+];
+
+// ...export statement
+```
+
+Our `Home` route is nested within our `App` route, and our `UserProfile` route
+is nested within our `Home` route.
+
+If we provide a piece of data to the `Outlet` component within our `App`, and we
+want to access it within our `UserProfile` component, we'll have to pass that
+data to the `Outlet` component within our `Home` component first.
+
+Essentially, `useOutletContext` only looks at the _immediate parent_ `Outlet`
+for data. So, if we have one `Outlet` nested within another `Outlet`, we'll need
+to make sure we pass data to that inner `Outlet` as well:
+
+```jsx
+// Home.js
+import { Outlet, useOutletContext } from "react-router-dom";
+import UserCard from "../components/UserCard";
+
+function Home(){
+    const users = useOutletContext();
+    const userList = users.map(user => <UserCard key={user.id} {...user}/>);
+
+  return (
+      <main>
+        <h1>Home!</h1>
+        <Outlet context={users}/>
+        {userList}
+      </main>
+  );
+};
+
+export default Home;
+```
+
+Now we can successfully access that data within our `UserProfile` component:
+
+```jsx
+// UserProfile.js
+import { useParams, useOutletContext } from "react-router-dom";
+import NavBar from "../components/Navbar";
+
+function UserProfile() {
+  const params = useParams();
+  const users = useOutletContext();
+
+  const user = users.find(user => user.id === parseInt(params.id));
+
+  return(
+      <aside>
```

Comment:
That was when the UserProfile was its own page. In this example, it will be rendered as a nested component, which means it's render as part of `Home`, which already has a `main` element. I believe accessibility best practices dictate that there should be only one `main` element per page.

If we think this needs some further explanation, I'm down to include some language about that! Could give us a chance to teach a little bit about accessibility / html best practices.

Reply:
Gotcha - makes sense. Using aside seems a little weird, though - this seems more like primary content than an aside. Is there something else that would work? article maybe?

Reply:
My motivation to make this an aside is that the main content for the Home page would be the list of available users. If someone clicked on one, it would open up their profile in an `aside` via the nested route. But we could definitely change to "article" or "section" if we think that's clearer from a semantic standpoint.

I'll go ahead and add in a little clarifying language and see if that helps.

Reply:
Ok, I've added the note and some minimal styling (literally just `float: right` for `aside`). If you want to see what that looks like, you can check out the solution branch. If we don't like it, I'll just revert to the previous commit.

Reply:
I like this change - it was looking a little like a mistake with the exact same thing rendered by UserProfile as was in the list, and I like the explanation you added. FYI at some point I'll probably add some more profile data to render in UserProfile to make it look a little more realistic. For now, though, I think this is in good shape!